xbmc/pvr/channels/PVRChannelGroups.cpp

```diff
@@ -240,7 +245,11 @@ bool CPVRChannelGroups::UpdateGroupsEntries(const CPVRChannelGroups &groups)
 
     // add it if not
     if (!existingGroup)
-      m_groups.push_back(CPVRChannelGroupPtr(new CPVRChannelGroup(m_bRadio, -1, (*it)->GroupName())));
+    {
+      existingGroup = *(m_groups.insert(m_groups.end(),CPVRChannelGroupPtr(new CPVRChannelGroup())));
```

Comment:
Yeah I see what you did there and it's completely valid, but it's not very readable ;)